common/common-api/src/main/java/org/ow2/proactive/scripting/Script.java

```diff
@@ -435,6 +429,17 @@ public boolean equals(Object o) {
         return false;
     }
 
+    /**
+     * Get MD5 hash value of the script without parameters
+     */
+    public static String digest(String script) {
+        try {
+            return new String(MessageDigest.getInstance("MD5").digest(script.getBytes()));
+        } catch (NoSuchAlgorithmException e) {
```

Comment:
It would be very complicated to throw an exception from this code. I could add a logger to Script eventually.